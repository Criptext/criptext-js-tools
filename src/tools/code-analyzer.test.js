/* eslint-env node, jest */

const analyze = require('./code-analyzer.js');

it('formats code correctly', () => {
  const code = `
  const helloWorld = "Hello World!"


  console.log(   helloWorld)
  `;

  const { prettierSourceCode, eslintReport } = analyze(code, 'helloWorld.js');

  expect(eslintReport).toBe(null);
  expect(prettierSourceCode).toMatchSnapshot();
});

it('catches eslint rule violations and presents them in a readable string', () => {
  const code = `let x = 1;

console.log(x);
`;

  const { prettierSourceCode, eslintReport } = analyze(code, 'sample.js');

  expect(typeof eslintReport).toEqual('string');
  expect(prettierSourceCode).toBe(null);
});
