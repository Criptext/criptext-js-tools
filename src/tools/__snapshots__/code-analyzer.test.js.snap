// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given a backend project formats code correctly 1`] = `
"const helloWorld = 'Hello World!';

console.log(helloWorld);
"
`;
